coffee/shepherd.coffee

```diff
@@ -117,6 +117,13 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+    if @options && @options.beforeShowPromise && typeof @options.beforeShowPromise is 'function'
+      @options.beforeShowPromise().then(=>
+        @_show()
+      )
+    else
+      @_show()
+  _show: =>
```

Comment:
Super nitpicky - can we keep the spacing consistent, and have a carriage return between function definitions?